openapi.yaml

```diff
@@ -45,6 +45,126 @@ tags:
       description: Find out more about Juju clouds.
       url: https://juju.is/docs/olm/manage-clouds
 paths:
+  /application/{name}:
+    get:
+      tags:
+        - application
+      description: Returns application information.
+      operationId: getApplication
+      parameters:
+        - $ref: "#/components/parameters/ApplicationName"
+      responses:
+        "200":
+          description: Detailed information about the requested application.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Application"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+    post:
+      tags:
+        - application
+      description: Is a one-stop deployment method for repository charms. Only a charm
+        name is required to deploy. If argument validation fails, a list of all errors
+        found in validation will be returned. If a local resource is provided, details
+        required for uploading the validated resource will be returned.
+      operationId: addApplication
+      parameters:
+        - $ref: "#/components/parameters/ApplicationName"
```

Comment:
This should be in the `requestBody` I think

Reply:
I may be misunderstanding, but I thought this was the identifier that we want the application to have after deploy, no? as in  `juju deploy ubuntu -->my-application<--`, cause the name of the charm to be used to deploy is already in the requestBody (see `charmName` in `DeployArgs`).

Reply:
Yep, but I think there is still some misunderstanding based on the latest commits. I don't think we should have a `POST /model{model-name}/application/{app-name}/deploy` - I think we should just have `POST /deploy` and have both the charm name and application name in the request body.

Reply:
Ah I see, that makes much more sense now 👍